src/kontrol/foundry.py

```diff
@@ -741,7 +741,7 @@ def foundry_to_xml(foundry: Foundry, proofs: list[APRProof]) -> None:
         tests += 1
         test, *_ = proof.id.split(':')
         contract, test_name = test.split('.')
-        _, contract_name = contract.split('%')
+        _, contract_name = contract.rsplit('%', 1)
```

Comment:
Since processing the `proof.id` or `test.id` to get the contract name and the test name is being done more than once and probably will be needed in upcoming fixes, I would suggest refactoring this into a separate function:
```python
def get_contract_test_names(id: str):
    test, *_ = id.split(':')
    contract, test_name = test.split('.')
    _, contract_name = contract.split('%')
    return contract_name, test_name
```

Reply:
I'm not sure that would work since `contract` would be discarded by that function but is used in the following line with `foundry_contract = foundry.contracts[contract]`